misc/metrics-extensions/modules/ballerina-metrics-extension/src/main/java/org/ballerinalang/observe/metrics/extension/defaultimpl/DefaultGauge.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinalang.observe.metrics.extension.defaultimpl;
+
+import org.ballerinalang.util.metrics.AbstractMetric;
+import org.ballerinalang.util.metrics.Gauge;
+import org.ballerinalang.util.metrics.MetricId;
+import org.ballerinalang.util.metrics.Snapshot;
+import org.ballerinalang.util.metrics.StatisticConfig;
+
+import java.util.concurrent.atomic.DoubleAdder;
+import java.util.concurrent.atomic.LongAdder;
+
+/**
+ * An implementation of {@link Gauge}.
+ */
+public class DefaultGauge extends AbstractMetric implements Gauge {
+
+    private static final RollingHistogram[] EMPTY_ROLLING_HISTOGRAMS = new RollingHistogram[0];
+    private static final Snapshot[] EMPTY_SNAPSHOTS = new Snapshot[0];
+
+    private final LongAdder count = new LongAdder();
+    private final DoubleAdder sum = new DoubleAdder();
+    private final DoubleAdder value = new DoubleAdder();
+    private final RollingHistogram[] rollingHistograms;
+
+    public DefaultGauge(MetricId id, Clock clock, StatisticConfig... statisticConfigs) {
+        super(id);
+        if (statisticConfigs != null) {
+            rollingHistograms = new RollingHistogram[statisticConfigs.length];
+            for (int i = 0; i < statisticConfigs.length; i++) {
+                rollingHistograms[i] = new RollingHistogram(clock, statisticConfigs[i]);
+            }
+        } else {
+            rollingHistograms = EMPTY_ROLLING_HISTOGRAMS;
+        }
+    }
+
+    public DefaultGauge(MetricId id, StatisticConfig... statisticConfigs) {
+        this(id, Clock.DEFAULT, statisticConfigs);
+    }
+
+
+    private void updateHistogram(double value) {
+        count.increment();
+        sum.add(value);
+        for (int i = 0; i < rollingHistograms.length; i++) {
```

Comment:
Shall we replace this with for each?

Reply:
Array index based access is better that for each.